ACE: Xtensa: Build failures on ACE platforms.

**Describe the bug**
Build Failures on `intel_adsp/ace_15_mtpm/sim, intel_adsp_ace_20_lnl/sim, intel_adsp/ace_30/ptl/sim.`

**To Reproduce**
`twister -p intel_adsp/ace15_mtl/sim -T tests/`

**Impact**
Affected ACE platforms.

**Logs and console output**
Summary:
Ace15_mtpm/sim
  ```
INFO    - 2357 test scenarios (2139 configurations) selected, 1960 configurations filtered (1880 by static filter, 80 at runtime).
INFO    - 0 of 179 executed test configurations passed (0.00%), 0 built (not run), 0 failed, 179 errored, with no warnings in 6773.04 seconds.
INFO    - 0 of 1876 executed test cases passed (0.00%), 1876 blocked on 1 out of total 871 platforms (0.11%).
INFO    - test configurations executed on platforms, 179 test configurations were only built.
  ```
Ace20_lnl/sim
```
INFO    - 2357 test scenarios (2139 configurations) selected, 1960 configurations filtered (1880 by static filter, 80 at runtime).
INFO    - 0 of 179 executed test configurations passed (0.00%), 0 built (not run), 0 failed, 179 errored, with no warnings in 6636.67 seconds.
INFO    - 0 of 1876 executed test cases passed (0.00%), 1876 blocked on 1 out of total 871 platforms (0.11%).
INFO    - 0 test configurations executed on platforms, 179 test configurations were only built.
```
Ace30/ptl/sim
```
INFO    - 2357 test scenarios (2139 configurations) selected, 1943 configurations filtered (1880 by static filter, 63 at runtime).
INFO    - 0 of 196 executed test configurations passed (0.00%), 0 built (not run), 0 failed, 196 errored, with no warnings in 7682.14 seconds.
INFO    - 0 of 2039 executed test cases passed (0.00%), 2039 blocked on 1 out of total 871 platforms (0.11%).
INFO    - 0 test configurations executed on platforms, 196 test configurations were only built.
```

All ACE platforms have the same issue.
```
2024-12-02T03:25:52.2953437Z [961/1107] Linking C static library modules/picolibc/libc.a
2024-12-02T03:25:52.2954551Z [962/1107] Generating include/generated/zephyr/syscall_dispatch.c, include/generated/zephyr/syscall_list.h
2024-12-02T03:25:52.2955948Z [963/1107] Building C object zephyr/CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj
2024-12-02T03:25:52.2957124Z FAILED: zephyr/CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj 
2024-12-02T03:25:52.2981959Z /opt/toolchains/xtensa/XtDevTools/install/tools/RI-2022.10-linux/XtensaTools/bin/xt-clang -DKERNEL -DK_HEAP_MEM_POOL_SIZE=0 -DTC_RUNID=bf5f30539d87dc1dd730de03d1669815 -D__LINUX_ERRNO_EXTENSIONS__ -D__ZEPHYR__=1 -I<path_to>/kernel/include -I<path_to>/arch/xtensa/include -I<path_to>/twister-out/intel_adsp_ace15_mtpm_sim/tests/kernel/mem_slab/mslab_concept/kernel.memory_slabs.concept/zephyr/include/generated/zephyr -I/<path_to>/include -I<path_to>/twister-out/intel_adsp_ace15_mtpm_sim/tests/kernel/mem_slab/mslab_concept/kernel.memory_slabs.concept/zephyr/include/generated -I<path_to>/soc/intel/intel_adsp -I<path_to>/soc/intel/intel_adsp/common -I<path_to>/soc/intel/intel_adsp/common/include -I<path_to>/soc/intel/intel_adsp/ace -I<path_to>/soc/intel/intel_adsp/ace/include -I<path_to>/soc/intel/intel_adsp/ace/include/ace15_mtpm -I<path_to>/subsys/testsuite/include -I<path_to>/subsys/testsuite/coverage -I<path_to>/subsys/testsuite/ztest/include -I<path_to>/drivers -isystem <path_to>/lib/libc/common/include -isystem <path_to>/twister-out/intel_adsp_ace15_mtpm_sim/tests/kernel/mem_slab/mslab_concept/kernel.memory_slabs.concept/modules/picolibc/picolibc/include -fno-strict-aliasing -Werror -Os -imacros <path_to>/twister-out/intel_adsp_ace15_mtpm_sim/tests/kernel/mem_slab/mslab_concept/kernel.memory_slabs.concept/zephyr/include/generated/zephyr/autoconf.h -fno-common -g -gdwarf-4 -fcolor-diagnostics -fms-extensions -imacros <path_to>/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-typedef-redefinition -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wno-sometimes-uninitialized -Wno-shift-overflow -Wno-missing-braces -Wno-self-assign -Wno-address-of-packed-member -Wno-unused-function -Wno-initializer-overrides -Wno-section -Wno-unknown-warning-option -Wno-unused-variable -Wno-format-invalid-specifier -Wno-gnu -Wno-tautological-compare -Werror=implicit-int -fno-asynchronous-unwind-tables -Wno-vla -fmacro-prefix-map=<path_to>/tests/kernel/mem_slab/mslab_concept=CMAKE_SOURCE_DIR -fmacro-prefix-map=<path_to>=ZEPHYR_BASE -fmacro-prefix-map=/__w/os.rtos.zephyr.zephyr=WEST_TOPDIR -ffunction-sections -fdata-sections -mlongcalls -std=c99 -fno-lto -MD -MT zephyr/CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj -MF zephyr/CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj.d -o zephyr/CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj -c <path_to>/arch/xtensa/core/offsets/offsets.c
2024-12-02T03:25:52.3004476Z In file included from <path_to>/arch/xtensa/core/offsets/offsets.c:6:
2024-12-02T03:25:52.3005899Z In file included from <path_to>/kernel/include/gen_offset.h:79:
2024-12-02T03:25:52.3007275Z In file included from <path_to>/include/zephyr/toolchain.h:42:
2024-12-02T03:25:52.3008654Z In file included from <path_to>/include/zephyr/toolchain/xcc.h:40:
2024-12-02T03:25:52.3010055Z In file included from <path_to>/include/zephyr/toolchain/llvm.h:31:
2024-12-02T03:25:52.3012031Z <path_to>/include/zephyr/toolchain/gcc.h:189:9: error: '__in_section_unique' macro redefined [-Werror,-Wmacro-redefined]
2024-12-02T03:25:52.3013637Z #define __in_section_unique(seg) ___in_section(seg, __FILE__, __COUNTER__)
2024-12-02T03:25:52.3014334Z         ^
2024-12-02T03:25:52.3015228Z <path_to>/include/zephyr/toolchain/xcc.h:20:9: note: previous definition is here
2024-12-02T03:25:52.3015468Z #define __in_section_unique(seg) \
2024-12-02T03:25:52.3015631Z         ^
2024-12-02T03:25:52.3016245Z In file included from <path_to>/arch/xtensa/core/offsets/offsets.c:6:
2024-12-02T03:25:52.3016806Z In file included from <path_to>/kernel/include/gen_offset.h:79:
2024-12-02T03:25:52.3017372Z In file included from <path_to>/include/zephyr/toolchain.h:42:
2024-12-02T03:25:52.3017944Z In file included from <path_to>/include/zephyr/toolchain/xcc.h:40:
2024-12-02T03:25:52.3018519Z In file included from <path_to>/include/zephyr/toolchain/llvm.h:31:
2024-12-02T03:25:52.3019600Z <path_to>/include/zephyr/toolchain/gcc.h:191:9: error: '__in_section_unique_named' macro redefined [-Werror,-Wmacro-redefined]
2024-12-02T03:25:52.3019835Z #define __in_section_unique_named(seg, name) \
2024-12-02T03:25:52.3019998Z         ^
2024-12-02T03:25:52.3020871Z <path_to>/include/zephyr/toolchain/xcc.h:23:9: note: previous definition is here
2024-12-02T03:25:52.3021117Z #define __in_section_unique_named(seg, name) \
2024-12-02T03:25:52.3021347Z         ^
2024-12-02T03:25:52.3021529Z 2 errors generated.
2024-12-02T03:25:52.3021760Z ninja: build stopped: subcommand failed.
```
**Environment:**
 - OS: Linux
 - Toolchain: xtensa
 - Commit SHA or Version used: v4.0.0-1264-g66df541d6c8d

**Additional context**
Bisect found.
```
commit 7e32b20
Author: Flavio Ceolin <flavio.ceolin@gmail.com>
Date:   3 days ago

    toolchain: xcc: Avoid undefs
    Re-order some definitions and header inclusion to avoid undef symbols.

    Signed-off-by: Flavio Ceolin <flavio.ceolin@gmail.com>
```

--- a/include/zephyr/toolchain/gcc.h
+++ b/include/zephyr/toolchain/gcc.h
@@ -186,10 +186,14 @@ do {                                                                    \
 				"." Z_STRINGIFY(c))))
 #define __in_section(a, b, c) ___in_section(a, b, c)
 
+#ifndef __in_section_unique
 #define __in_section_unique(seg) ___in_section(seg, __FILE__, __COUNTER__)
+#endif
 
+#ifndef __in_section_unique_named
 #define __in_section_unique_named(seg, name) \
 	___in_section(seg, __FILE__, name)
+#endif
 
 /* When using XIP, using '__ramfunc' places a function into RAM instead
  * of FLASH. Make sure '__ramfunc' is defined only when

